src/nvim/bufwrite.c

```diff
@@ -355,9 +355,11 @@ static int check_mtime(buf_T *buf, FileInfo *file_info)
       && time_differs(file_info, buf->b_mtime_read, buf->b_mtime_read_ns)) {
     msg_scroll = true;  // Don't overwrite messages here.
     msg_silent = 0;     // Must give this prompt.
-    // Don't use emsg() here, don't want to flush the buffers.
-    msg(_("WARNING: The file has been changed since reading it!!!"), HL_ATTR(HLF_E));
-    if (ask_yesno(_("Do you really want to write to it"), true) == 'n') {
+    if (vim_dialog_yesno(VIM_WARNING,
+                         NULL,
+                         _(
+                          "WARNING: The file has been changed since reading it. Do you really want to write to it?"),
```

Comment:
Just out of curiosity, any special reason to change the message slightly (e.g. dropping the exclamation marks)?

Reply:
No special reason, but I always found the three exclamation marks a bit over the top in addition to the all caps WARNING. Don't feel too strongly about it though.